pkg/apis/clusterlink.net/v1alpha1/accesspolicy.go

```diff
@@ -35,20 +34,22 @@ type PrivilegedAccessPolicy struct {
 }
 
 // +kubebuilder:object:root=true
-// +kubebuilder:resource:scope=Namespaced
+// +kubebuilder:resource:scope=Cluster
 
-// AccessPolicy defines whether a group of potential connections should be allowed or denied.
-// If multiple AccessPolicy objects match a given connection, privileged policies
-// take precedence over non-privileged, and within each tier deny policies take
-// precedence over allow policies.
-type AccessPolicy struct {
+// PrivilegedAccessPolicy is the cluster-scoped version of AccessPolicy.
+// PrivilegedAccessPolicies are intended to be used by cluster admins, and take precedence over AccessPolicies.
+// Within each tier, deny policies take precedence over allow policies.
+type PrivilegedAccessPolicy struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
 	// Spec represents the attributes of the exported service.
 	Spec AccessPolicySpec `json:"spec,omitempty"`
 }
 
+// Important note: In the code, PrivilegedAccessPolicy is unmarshalled into AccessPolicy.
+// The two structs must be therefore identical, or something may break.
```

Comment:
```suggestion
// Therefore, the two structs must be identical, or something may break.
```

Reply:
I wonder if there is some type assertion / cast that you can do to validate it in code so the compiler complains. I've seen it done for structs that implement an interface and must continue to do so as the interface evolves

Reply:
Introduced a new opaque, PDP-internal, generalized representation of AccessPolicy and PrivilegedAccessPolicy CRDs.
We no longer need this assertion